src/libs/types-rs/src/mount.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// Prefix to mark a volume as Kata special.
+pub const KATA_VOLUME_TYPE_PREFIX: &str = "kata:";
+
+/// The Mount should be ignored by the host and handled by the guest.
+pub const KATA_GUEST_MOUNT_PREFIX: &str = "kata:guest-mount:";
+
+/// KATA_EPHEMERAL_DEV_TYPE creates a tmpfs backed volume for sharing files between containers.
+pub const KATA_EPHEMERAL_VOLUME_TYPE: &str = "kata:ephemeral";
+
+/// KATA_HOST_DIR_TYPE use for host empty dir
+pub const KATA_HOST_DIR_VOLUME_TYPE: &str = "kata:hostdir";
+
+/// Information about a mount.
+#[derive(Debug, Clone, Default, serde::Serialize, serde::Deserialize)]
+pub struct Mount {
+    /// The source path for the mountpoint.
+    pub source: String,
+    /// The destination path for the mountpoint.
+    pub destination: String,
+    /// The type of filesystem for the mountpoint.
+    pub fs_type: String,
+    /// Mount options for the mountpoint.
+    pub options: Vec<String>,
+    /// Device id for device associated with the mountpoint.
+    pub device_id: String,
+    /// Host side bind mount path for the mountpoint.
+    pub host_path: Option<String>,
+    /// Whether to mount the mountpoint in readonly mode
+    pub read_only: bool,
+}
+
+impl Mount {
+    /// Get size of mount options.
+    pub fn option_size(&self) -> usize {
+        self.options.iter().map(|v| v.len() + 1).sum()
+    }
+}
+
+/// Check whether a mount type is a marker for Kata specific volume.
+pub fn is_kata_special_volume(ty: &str) -> bool {
+    ty.len() > KATA_VOLUME_TYPE_PREFIX.len() && ty.starts_with(KATA_VOLUME_TYPE_PREFIX)
```

Comment:
I don't think you need the length checks here?

```suggestion
    ty.starts_with(KATA_VOLUME_TYPE_PREFIX)
```

Same comment for `is_kata_guest_mount_volume()`.

Reply:
It's some sort of minor optimizations, it may be removed for simplicity.

Reply:
Sorry, my bad:(
"ty.len() > KATA_VOLUME_TYPE_PREFIX.len()" is to ensure is_kata_special_volume() returns false for "kata:".

Reply:
I'd add a comment for that.

Reply:
Also.. why do you need to do that?

Reply:
Actually the KATA_VOLUME_TYPE_PREFIX acts as an marker for extensions, which basically says that "hi, this mount has been handled by some other plugins, please skip it."

Reply:
I know this is not optimal, but we have no mechanism to extend the oci::Mount structure yet.